intake_geopandas/geopandas.py

```diff
@@ -70,28 +83,75 @@ def __init__(self, urlpath, bbox=None,
         self._bbox = bbox
         self._geopandas_kwargs = geopandas_kwargs or {}
         self._dataframe = None
+        self.storage_options = storage_options or {}
+
+        # warn if using fsspec caching and same_names not True for zip files
+        if 'cache' in self.urlpath and 'zip' in self.urlpath:
+            same_names = False  # default
+            # find different same_names setting
+            for c in ['blockcache', 'filecache', 'simplecache']:
+                if c in self.storage_options:
+                    if 'same_names' in self.storage_options[c]:
+                        same_names = self.storage_options[c]['same_names']
+            if not same_names:
+                warnings.warn(
+                    'Need same_names = True for local caching of `zip` files.'
+                )
 
         super().__init__(metadata=metadata)
 
     def _open_dataset(self):
         """
         Open dataset using geopandas and use pattern fields to set new columns.
         """
+        if 'cache' in self.urlpath:  # new caching
+            url = fsspec.open_local(self.urlpath, **self.storage_options)
+            if not os.path.exists(url):
+                # explicitly loading the file if not cached
+                url = fsspec.open(self.urlpath, **self.storage_options)
```

Comment:
I dont understand why `url = fsspec.open_local(self.urlpath, **self.storage_options)` is not enough. in `intake-xarray` it is.

Reply:
#17 now only requires `fsspec.open_local`